lab-yunjoo/lib/storage.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const fs = require('fs');
+const Storage = module.exports = function(dataDir){
+  this.dataDir = dataDir;
+};
+
+Storage.prototype.setItem = function(schema, item){
+  return new Promise((resolve, reject)=>{
```

Comment:
This is where you should be using mkdirp to ensure that a schema directory exists before you try writing to it.